subsys/mgmt/osdp/src/osdp_cp.c

```diff
@@ -137,42 +140,60 @@ static int cp_build_command(struct osdp_pd *pd, uint8_t *buf, int max_len)
 
 	switch (pd->cmd_id) {
 	case CMD_POLL:
-		assert_buf_len(CMD_POLL_LEN, max_len);
+		if (!check_buf_len(CMD_POLL_LEN, max_len)) {
```

Comment:
Is this an unrecoverable error ? If not you should use `LOG_WRN` imho.
Also, `check_buf_len` is essentially just a branch. Wouldn't be better do the check here, and if it is false use a goto and log it with more information ? Something like:
```
case CMD_POLL:
        if (CMD_POLL_LEN > max_len) {
                goto err;
        }
.....


error:
	LOG_WRN("OOM at build command (%d) fails\n", pd->cmd_id);
	return OSDP_CP_ERR_GENERIC;
```

Reply:
This error should technically be unreachable since the both peers agree on TX/RX buffer size. If we _do_ hit, we should treat it as unrecoverable error (PD goes offline).

> Wouldn't be better do the check here, and if it is false [..]

When the method is `static inline`-ed. An optimising compiler should produce the same effect IIRC. Also, logging `need` and `have` is more important (than the command/reply ID as they are logged earlier).